docs/dev/building-on-zksync/contracts/differences-with-ethereum.md

```diff
@@ -0,0 +1,192 @@
+
+```solidity
+
+MyContract a = new MyContract();
+MyContract a = new MyContract{salt: ...}();
+
+```
+
+The following should also work as expected, but must be explicitly tested by the team in order to make sure that this is the case:
+
+```solidity
+bytes memory bytecode = type(MyContract).creationCode;
+assembly {
+    addr := create2(0, add(bytecode, 32), mload(bytecode), salt)
+} 
+```
+
+The following code will not work, because the compiler will not be aware of the bytecode beforehand:
+
+```solidity
+function myFactory(bytes memory bytecode) public {
+   assembly {
+      addr := create(0, add(bytecode, 0x20), mload(bytecode))
+   }
+}
+```
+
+Unfortunately it is not possible to distinguish between the cases above in compile time. And so it *highly recommended* to include tests for your factory regarding deploying the child contracts. 
+
+Also, note that for zkEVM bytecode we contain the different address derivation from Ethereum. The exact formulas are available in our SDK:
+
+```typescript
+export function create2Address(sender: Address, bytecodeHash: BytesLike, salt: BytesLike, input: BytesLike) {
+    const prefix = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('zksyncCreate2'));
+    const inputHash = ethers.utils.keccak256(input);
+    const addressBytes = ethers.utils
+        .keccak256(ethers.utils.concat([prefix, ethers.utils.zeroPad(sender, 32), salt, bytecodeHash, inputHash]))
+        .slice(26);
+    return ethers.utils.getAddress(addressBytes);
+}
+
+export function createAddress(sender: Address, senderNonce: BigNumberish) {
+    const prefix = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('zksyncCreate'));
+    const addressBytes = ethers.utils
+        .keccak256(
+            ethers.utils.concat([
+                prefix,
+                ethers.utils.zeroPad(sender, 32),
+                ethers.utils.zeroPad(ethers.utils.hexlify(senderNonce), 32)
+            ])
+        )
+        .slice(26);
+
+    return ethers.utils.getAddress(addressBytes);
+}
+```
+
+### Differences in `block.timestamp` and `block.number`
+These variables behave differently from L1. Read more [about blocks in zkSync](../../developer-guides/transactions/blocks.md#blocks-in-zksync-era).
+
+### Differences in `COINBASE` `DIFFICULTY` `BASEFEE` 
+- `COINBASE` returns our Bootloader contract address, `0x0000000000000000000000000000000000008001` 
+- `DIFFICULTY` returns a constant value of `2500000000000000` 
+- `BASEFEE` is not a constant, but is defined by the fee model. Most of the time it is 0.25 gwei, but under very high L1 gas prices it may rise.
+
+### Use of `codesize`/`codecopy` and `calldata` in the deploy code
+`Calldata` in the constructor is not empty as on the EVM, on zkEVM calldata contains constructor arguments.
+`codesize` has the same behavior as `calldatasize`, `codecopy` as `calldatacopy`.
+To make it works datasize of the current object is 0.
```

Comment:
I think it makes sense to add one more topic (or current should be extended):
### Using `return` in the deploy code will not function as intended 
Constructors on the zkSync Era return immutables arrays. If you attempt to use return, it will instead return the array of immutables in the auxiliary heap that was previously written there (using `setimmutable`) rather than the data you specified.